mne/viz/evoked.py

```diff
@@ -1526,18 +1527,38 @@ def _setup_styles(conditions, styles, cmap, colors, linestyles):
         cmapper = getattr(plt.cm, cmap, cmap)
         color_conds = list(colors.keys())
         all_colors = [colors[cond] for cond in color_conds]
-        n_colors = len(all_colors)
         color_order = np.array(all_colors).argsort()
         color_indices = color_order.argsort()
 
+        if all([isinstance(color, Integral) for color in all_colors]):
+            msg = "Integer colors detected, mapping to rank positions ..."
+            n_colors = len(all_colors)
+            colors_ = {cond: ind for cond, ind in
+                       zip(color_conds, color_indices)}
+
+            def convert_colors(color):
+                return colors_[color]
+        else:
+            for color in all_colors:
+                if not 0 <= color <= 1:
+                    raise ValueError("Values of colors must be all-integer or "
+                                     "floats between 0 and 1, got %s." % color)
+            msg = "Float colors detected, mapping to percentiles ..."
+            n_colors = 101  # percentiles plus 1 if we have 1.0s
+            colors_old = colors.copy()
+
+            def convert_colors(color):
+                return int(colors_old[color] * 100)
+            colors_are_float = True
```

Comment:
you are already setting `msg` inside these branches, so might as well also set `colors_are_float` in each branch for consistency (i.e., move the `colors_are_float = True` from 1518 up here

Reply:
I also need it for the `cmap=None` case, 1519.

Reply:
ahh okay